### PR TITLE
7903625: Simplify API of the generated code

### PR DESCRIPTION
**Pull request by mcimadamore**
This PR simplifies the API of the code generated by jextract. Here's a list of the changes implemented:

* the struct layout accessor is now just named `layout()`
* functional pointer code generation is different now: we generate a toplevel _class_ which defines a nested functional interface that is used for the factory method. This means that we can hide some of the constants associated with the function pointer. The descriptor accessor remains, but is called `descriptor()`.
* For global variables, accessors such as `x$SEGMENT` and `x$LAYOUT` were made `private`
* As for struct fields, I've made the accessors for global variable work for the case where the global is a struct/array too (the getter/setter will work on a MemorySegment).

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed (no review required)

### Issue
 * [CODETOOLS-7903625](https://bugs.openjdk.org/browse/CODETOOLS-7903625): Simplify API of the generated code (**Enhancement** - P3)

### Reviewers
 * [Jorn Vernee](https://openjdk.org/census#jvernee) (@JornVernee - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jextract.git pull/187/head:pull/187` \
`$ git checkout pull/187`

Update a local copy of the PR: \
`$ git checkout pull/187` \
`$ git pull https://git.openjdk.org/jextract.git pull/187/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 187`

View PR using the GUI difftool: \
`$ git pr show -t 187`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jextract/pull/187.diff">https://git.openjdk.org/jextract/pull/187.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jextract/pull/187#issuecomment-1900390720)

### CODE REVIEW

**Review by JornVernee — APPROVED** [review]
This looks good. Could you add a test case for the new setter as well?

File: test/jtreg/generator/structAccessors/TestNestedStructAccess.java [hunk 1]
```diff
@@ -33,7 +33,7 @@
 /*
  * @test
   * @library /lib
- * @run main/othervm JtregJextract -l Func -t test.jextract.nestedaccess nestedStructAccess.h
+ * @run main/othervm JtregJextract -t test.jextract.nestedaccess nestedStructAccess.h
```
Comment (mcimadamore) [inline test/jtreg/generator/structAccessors/TestNestedStructAccess.java:36]: the `-l` flag seems to be not required here

File: test/jtreg/generator/structAccessors/TestGlobalStructAccess.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import static test.jextract.globalaccess.globalStructAccess_h.*;
+
+/*
+ * @test
+ * @library /lib
+ * @run main/othervm JtregJextract -l StructGlobal -t test.jextract.globalaccess globalStructAccess.h
+ * @build TestGlobalStructAccess
+ * @run testng/othervm --enable-native-access=ALL-UNNAMED TestGlobalStructAccess
+ */
+public class TestGlobalStructAccess {
```
Comment (mcimadamore) [inline test/jtreg/generator/structAccessors/TestGlobalStructAccess.java:40]: I've renamed the existing test in this folder, which was also checking struct getter/setter, but nested inside other structs.

**Review by JornVernee — APPROVED** [review]

**Pull request closed without merging**